src/averageAges.js

```diff
@@ -55,7 +74,26 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const mothers = people.filter(p => p.sex === 'f'
+    && (people.some(m => m.mother === p.name)));
+
+  const children = people
+    .filter(child => mothers.some(mother => child.mother === mother.name));
+
+  const withSon = onlyWithSon
+    ? children.filter(child => child.sex === 'm')
+    : children;
```

Comment:
you may add an additional condition inside the first filtering (inside the nested `some` method), similar as you do with `century` in the first function